### DISCUSSION TITLE
tutorials/voyage-part3/06-schema-checks-in-ci

### DISCUSSION THREAD

**Discussion by giscus** [apollographql-education/content-feedback#93] [course feedback]
# tutorials/voyage-part3/06-schema-checks-in-ci

https://www.apollographql.com/tutorials/voyage-part3/06-schema-checks-in-ci

**Comment by torressam333** [comment]
If we are not using a federated graph, how can we use Rover to only run checks on operations?

**Reply by lizhennessy** [reply]
Hi @torressam333! Here's a great article on [using Rover with a monograph](https://www.apollographql.com/docs/rover/commands/graphs/) (non-federated). There's a few commands here that could be helpful!

**Comment by rickyssj2** [comment]
If the Github Actions is also using the same rover subgraph check command behind the scenes, isn't it redundant to have a local and an automated check? 
Will there be a scenario that the local check passes but the Github triggered check fails?

**Reply by lizhennessy** [reply]
Hi @rickyssj2 ! Great question. In one sense, we can think of running `rover subgraph check` as our personal assurance against the current state of the graph that the changes we've made won't break anything. But one of the nice things about building a federated graph is that we can do so as part of a much larger team; this means that lots of developers can be working simultaneously on different features of the graph, shipping them incrementally, and running `rover subgraph check` in their own environment.

We could imagine a scenario where we've finished up our changes, run `rover subgraph check`, and confirm that our changes are compatible with the existing state of the graph. We might not merge for a couple of days, however! (Code review can take awhile! 😆 ) And in that time, our graph might have changed; other teams could ship different pieces of the schema that suddenly conflict with ours. You can think of running `rover subgraph check` _locally_ as our way of initially validating our changes. However, we should also have it integrated into CI/CD so that it's run _again_ right before the changes are published as part of our graph. These sneaky conflicts might not occur that frequently on a small team, but as our graph grows we can be sure that the schema is constantly evolving! I hope this helps answer your question. Do let us know if there's anything else we can assist with.
